packages/react-router/__tests__/navigate-test.tsx

```diff
@@ -115,4 +116,65 @@ describe("navigate", () => {
       expect(node.innerHTML).toMatchInlineSnapshot(`"<h1>About</h1>"`);
     });
   });
+
+  describe("with a search param", () => {
+    it("navigates to the correct URL with params", () => {
+      function Home() {
+        let navigate = useNavigate();
+
+        function handleClick() {
+          navigate({
+            pathname: "../about",
+            search: new URLSearchParams({ user: "mj" }).toString()
+          });
+        }
+
+        return (
+          <div>
+            <h1>Home</h1>
+            <button onClick={handleClick}>click me</button>
+          </div>
+        );
+      }
+
+      function About() {
+        let { search } = useLocation();
+        let { user } = qsParse(search);
```

Comment:
This could be:

```ts
let user = new URLSearchParams(useLocation().search).get("user");
```

No need to roll your own `qsParse` function. `URLSearchParams` is capable of parsing URL query strings directly.